include/sway/input/seat.h

```diff
@@ -185,6 +185,10 @@ struct sway_workspace *seat_get_last_known_workspace(struct sway_seat *seat);
 
 struct sway_container *seat_get_focused_container(struct sway_seat *seat);
 
+// Force focus to a particular surface that is not part of the workspace
+// hierarchy (used for lockscreen)
+void sway_force_focus(struct wlr_surface *surface);
```

Comment:
Can't we re-use the wlr-input-inhibitor implementation stuff for this?

Reply:
I mostly did (that interface wires up to `seat_set_exclusive_client` which the lock uses to block input to other clients) but that doesn't handle setting the focus to the client in the first place.  Normally, that's because a client already has focus when it requests the inhibitor be activated.

Reply:
Note: at least swaylock v1.6 will run `lock` before creating any surfaces at all.  That requires setting up the inhibitor right away (which works just fine, dropping any input except `--locked` binds).  However, the `wl_keyboard` API seems to require a surface for input (and all the helpers that `sway_force_focus` calls require it too), so we need to postpone that until we have such a surface.